### PR TITLE
[storage] Protect against nil Helper and/or Handler in BalanceStorage

### PR DESCRIPTION
**Pull request by patrick-ogrady**
Someone was able to [force a panic in `rosetta-cli` because of an unpopulated BalanceStorage helper and handler](https://community.rosetta-api.org/t/operation-status-must-be-empty-for-construction/315/4?u=patrick.ogrady). This PR causes BalanceStorage to raise an error in these cases instead of panicking.

### Changes
- [x] Add new error
- [x] Check for nil helper or handler
- [x] Add new tests

### CODE REVIEW

File: storage/modules/balance_storage.go [hunk 1]
```diff
@@ -187,6 +187,10 @@ func (b *BalanceStorage) AddingBlock(
 	block *types.Block,
 	transaction database.Transaction,
 ) (database.CommitWorker, error) {
+	if b.handler == nil {
```
Comment (septerr) [inline storage/modules/balance_storage.go:190]: Can we force the handler to be present in the methods used to create a new BalanceStorage instance?

Comment (patrick-ogrady) [reply storage/modules/balance_storage.go:190]: I wish! There is a somewhat complex dependency process where we can't add the handler or helper during initialization.

Comment (patrick-ogrady) [reply storage/modules/balance_storage.go:190]: We need a reference to `BalanceStorage` to create some useful helpers (we should probably clean this up): https://github.com/coinbase/rosetta-cli/blob/972a1690001f7f7d22fafa0a580512eac8517add/pkg/tester/data.go#L284-L305

Comment (septerr) [reply storage/modules/balance_storage.go:190]: yeah, we should def clean it up!

**Review by septerr — APPROVED** [review]

**Pull request merged**